runtime/commonMain/src/kotlinx/serialization/Annotations.kt

```diff
@@ -10,60 +10,156 @@ import kotlinx.serialization.json.*
 import kotlin.reflect.*
 
 /**
- * Sets specific serializer for class, property or type argument.
- * When argument is omitted, plugin will generate default implementation inside the class.
+ * The main entry point to the serialization process.
+ * Applying [Serializable] to the Kotlin class instructs the serialization plugin to automatically generate [KSerializer]
+ * for the current class, that can be used to serialize and deserialize the class.
+ * The generated serializer can be accessed with `.serializer()` extension method that is generated by the plugin as well.
```

Comment:
better mention that this is a companion/type extension method because you do not have to have a class instance.

Also, isn't it better to use 'function' instead of 'method'?